variables.tf

```diff
@@ -160,21 +164,27 @@ variable "capacity_type" {
   }
 }
 
-variable "block_device_mappings" {
-  type        = list(any)
+variable "block_device_map" {
+  type = map(object({
+    no_device    = optional(bool, null)
+    virtual_name = optional(string, null)
+    ebs = optional(object({
+      delete_on_termination = optional(bool, true)
+      encrypted             = optional(bool, true)
+      iops                  = optional(number, null)
+      kms_key_id            = optional(string, null)
+      snapshot_id           = optional(string, null)
+      throughput            = optional(number, null)
+      volume_size           = optional(number, 20)
+      volume_type           = optional(string, "gp3")
+    }))
+  }))
+
   description = <<-EOT
-    List of block device mappings for the launch template.
-    Each list element is an object with a `device_name` key and
-    any keys supported by the `ebs` block of `launch_template`.
+    Map of block device name specification, see [launch_template.block-devices](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#block-devices).
     EOT
   # See https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#ebs
-  default = [{
-    device_name           = "/dev/xvda"
-    volume_size           = 20
-    volume_type           = "gp2"
-    encrypted             = true
-    delete_on_termination = true
-  }]
+  default = { "/dev/xvda" = { ebs = {} } }
```

Comment:
Suggestion: Add known operating systems and their default storage in description notes